### DISCUSSION TITLE
Fractional zoom levels

### DISCUSSION THREAD

**Discussion by vrtxd** [godotengine/godot-proposals#6165] [Editor]
*The issue is relevant to all Godot projects that make use of the 2D viewport. See also discussion: Zoom steps weird? (#6082)*

### Problem

Currently in Godot's 2D viewport, zoom levels that are changed with the mouse wheel seem to be calculated with a constant ratio of the twelfth root of two. While this irrational number does make the relationship between zoom levels mathematically consistent, it results in unintuitive percentages while disregarding very common fraction-based zoom levels such as 66%, 75%, 125%, 133%, 150%, 167% and 175%. Changing the zoom with the +/– buttons in the GUI changes the zoom level by a factor of square root of 2.

### Solution

Preserving the current 12-step interval but rounding the zoom factors to a nearby common fraction, I propose a new implementation of zoom levels that I believe is more intuitive for users/developers/designers/creators.

### Mockup

```
    ...
	89.1% --> 87.50% (7/8)
	94.4% --> 93.75% (15/16)

    100.00% (1/1)
	105.9% --> 106.25% (17/16)
	112.2% --> 112.50% (9/8)
	118.9% --> 120.00% (6/5)
	126.0% --> 125.00% (5/4)
	133.5% --> 133.33% (4/3)
	141.4% --> 140.00% (7/5)
	149.8% --> 150.00% (3/2)
	158.7% --> 160.00% (8/5)
	168.2% --> 166.67% (5/3)
	178.2% --> 175.00% (7/4)
	188.8% --> 187.50% (15/8, i.e. 15/16 × 2)
	200.00% (2/1, i.e. 1/1 × 2)

	211.9% --> 212.50% (17/8, i.e. 17/16 × 2)
	224.5% --> 225.00% (9/4, i.e. 9/8 × 2)
	...
```

**If this enhancement will not be used often, can it be worked around with a few lines of script?**

To my understanding, the zoom function is used pretty often. I am not a programmer, but I think this feature does not require more than a few lines of script.

**Is there a reason why this should be core and not an add-on in the asset library?**

I believe this feature is a basic quality of life design improvement to the editor's overall user experience, rather than just a personal customization preference that users might optionally want to opt into.

**Comment by AThousandShips** [comment]
From the editor code: "This allow a smooth geometric evolution of the zoom, with the advantage of visiting all integer power of two scale factors."

**Reply by vrtxd** [reply]
This new system would also allow a smooth geometric evolution of the zoom, with the advantage of visiting all integer power of two scale factors: 100%, 200%, 400%, 800%, 1600% etc. The difference is that this proposal would also make the 11 values between those powers of two more usable. Current system provides no way of zooming to exact intermediate fractional values like 150%, 175%, 225%, 250%, 300%, 350%, 450%, 500%, 600%, 750%, 900%, 1000%, 1200% etc.
